.github/workflows/smart-contracts-integrations-tests-ghcr.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and Publish Smart Contracts Integration Tests image to GHCR Private Repository
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "Tag for GHCR image"
+        required: true
+        type: string
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          logout: true
+
+      - name: Build and Push Docker Image
+        uses: docker/build-push-action@v4
+        with:
+          context: toolkit/offchain/tests/docker/
+          file: toolkit/offchain/tests/docker/Dockerfile
+          push: true
+          tags: 
+            ghcr.io/partner-chains/smart-contracts-integration-tests:${{ inputs.tag }}
```

Comment:
In my draft PR #307 I've proposed `partner-chains-smart-contracts-tests-cardano-node-ogmios`, it is verbose but strongly suggests what it is.

Because `smart-contracts` is in the repository part, I think `smart-contracts-tests-cardano-node-ogmios` would fit nicely

Reply:
`smart-contracts` is not in the repository part if we are publishing this to `partner-chains` repository. Is this definitely supposed to be published under `partner-chains` repo GHCR?

Reply:
Yes, my bad. I wanted to say:
Because `partner-chains` is in the repository part, I think `smart-contracts-tests-cardano-node-ogmios` would fit nicely.

Sorry

Reply:
Oh I think you meant `partner-chains` is already in the repository

Updated to `smart-contracts-tests-cardano-node-ogmios` package name